nova-act/usecases/price_comparison/main.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2025 Amazon Inc
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Compare product prices across multiple retailers.
+
+Searches for a product across Amazon, Best Buy, Costco, and Target concurrently
+and displays pricing and promotion details in a comparison table.
+
+Usage:
+    python main.py [--product_name <product name>] [--product_sku <sku>] [--sources <sources>] [--headless]
+    python main.py --product_name "iPad Pro 13-inch, 256GB Wi-Fi" --product_sku "MVX23LL/A"
+    python main.py --sources '[("Walmart", "https://www.walmart.com"), ("Amazon", "https://www.amazon.com")]'
+    python main.py --headless=False
+"""
+
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+import fire
+import pandas as pd
+from nova_act import BOOL_SCHEMA, ActAgentError, NovaAct
+from pydantic import BaseModel
+
+# Global defaults
+DEFAULT_PRODUCT_NAME = "iPad Pro 13-inch (M4 chip), 256GB Wi-Fi"
+DEFAULT_PRODUCT_SKU = "MVX23LL/A"
+DEFAULT_PRODUCT_SOURCES = [
+    ("Amazon", "https://www.amazon.com"),
+    ("Best Buy", "https://www.bestbuy.com"),
+    ("Costco", "https://www.costco.com"),
+    ("Target", "https://www.target.com"),
+]
+
+
+class ProductPricing(BaseModel):
+    product_name: str | None
+    price: float | None
+    promotion_details: str | None
+    source: str  # Target, Best Buy, or Costco
+
+
+def check_source_price(
+    product_name: str,
+    product_sku: str,
+    source: str,
+    starting_url: str,
+    headless: bool = True,
+) -> ProductPricing | None:
+    """Check price for a product at a specific source."""
+
+    try:
+        with NovaAct(starting_page=starting_url, headless=headless) as nova:
+            # Check for captcha
+            captcha_check = nova.act(
+                "Is there a captcha on the screen?", schema=BOOL_SCHEMA
+            )
+            if captcha_check.matches_schema and captcha_check.parsed_response:
+                input(f"Please solve the captcha for {source} and hit return when done")
+
+            # Search using product SKU
+            nova.act(f"Search for '{product_sku}'.")
+
+            # Extract pricing and details from the most relevant result based on product name
+            result = nova.act(
```

Comment:
you might have to change this API call to nova.act_get(). Have you tested this lately?